CS4125Project/src/Services/DetailedReceiptBuilder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package Services;
+
+import java.util.ArrayList;
+
+/**
+ *
+ * @author Jack
+ */
+public class DetailedReceiptBuilder implements ReceiptBuilder{
+    private Receipt receipt;
+    
+    public DetailedReceiptBuilder(){
+        this.receipt = new Receipt();
+    }
+    
+    public void buildusername(){
+        receipt.SetUsername(Shop.getInstance().getAccount().getUsername());
+    }
+    
+    public void buildEmail(){
+        receipt.SetEmail(Shop.getInstance().getAccount().getEmail());
+    }
+    
+    public void buildPaymentDetails(){
+        receipt.SetPaymentDetails(Shop.getInstance().getAccount().getPaymentType().getPaymentInfo());
+    }
+    
+    public void bulidDelivery(){
+        receipt.SetDeliveryMethod(Shop.getInstance().getAccount().getLocation());
+    } 
+  
+    public void buildPurchases(){
+        ArrayList<Purchase> basket = new ArrayList<Purchase>();
+        basket = ShoppingBasket.GetInstance().GetBasketContents();
```

Comment:
Unnecessary extra assignment on first line

You could just say
```java
ArrayList<Purchase> basket = ShoppingBasket.GetInstance().GetBasketContents();
```
Same with other concrete builder classes